--- ../vdr-2.2.0.plain//menu.c	2015-02-10 13:37:06.000000000 +0100
+++ menu.c	2017-02-17 10:06:59.655147352 +0100
@@ -1302,6 +1302,7 @@
 {
   cOsdMenu::Display();
   DisplayMenu()->SetEvent(event);
+  cStatus::MsgOsdSetEvent(event);
   if (event->Description())
      cStatus::MsgOsdTextItem(event->Description());
 }
@@ -1432,6 +1433,7 @@
   static void SetCurrentChannel(int ChannelNr) { currentChannel = ChannelNr; }
   static const cEvent *ScheduleEvent(void);
   virtual eOSState ProcessKey(eKeys Key);
+  virtual void Display(void);
   };
 
 int cMenuWhatsOn::currentChannel = 0;
@@ -1461,6 +1463,18 @@
   SetHelpKeys();
 }
 
+void cMenuWhatsOn::Display(void)
+{
+   cOsdMenu::Display();
+   
+   if (Count() > 0) {
+      int ni = 0;
+      for (cOsdItem *item = First(); item; item = Next(item)) {
+         cStatus::MsgOsdEventItem(((cMenuScheduleItem*)item)->event, item->Text(), ni++, Count());
+      }
+   }
+}
+
 bool cMenuWhatsOn::Update(void)
 {
   bool result = false;
@@ -1614,6 +1628,7 @@
   cMenuSchedule(void);
   virtual ~cMenuSchedule();
   virtual eOSState ProcessKey(eKeys Key);
+  virtual void Display(void);
   };
 
 cMenuSchedule::cMenuSchedule(void)
@@ -1640,6 +1655,18 @@
   cMenuWhatsOn::ScheduleEvent(); // makes sure any posted data is cleared
 }
 
+void cMenuSchedule::Display(void)
+{
+   cOsdMenu::Display();
+   
+   if (Count() > 0) {
+      int ni = 0;
+      for (cOsdItem *item = First(); item; item = Next(item)) {
+         cStatus::MsgOsdEventItem(((cMenuScheduleItem*)item)->event, item->Text(), ni++, Count());
+      }
+   }
+}
+
 void cMenuSchedule::PrepareScheduleAllThis(const cEvent *Event, const cChannel *Channel)
 {
   Clear();
@@ -2561,6 +2588,7 @@
      }
   cOsdMenu::Display();
   DisplayMenu()->SetRecording(recording);
+  cStatus::MsgOsdSetRecording(recording);
   if (recording->Info()->Description())
      cStatus::MsgOsdTextItem(recording->Info()->Description());
 }
--- ../vdr-2.2.0.plain//menuitems.c	2015-02-09 12:53:10.000000000 +0100
+++ menuitems.c	2017-02-20 09:11:14.088587593 +0100
@@ -39,6 +39,7 @@
   cString buffer = cString::sprintf("%s:\t%s", name, Value);
   SetText(buffer);
   cStatus::MsgOsdCurrentItem(buffer);
+  cStatus::MsgOsdCurrentItemIndex(buffer, -1);
 }
 
 void cMenuEditItem::SetHelp(const char *Red, const char *Green, const char *Yellow, const char *Blue)
--- ../vdr-2.2.0.plain//osdbase.c	2015-01-15 11:11:11.000000000 +0100
+++ osdbase.c	2017-02-20 09:15:05.359961264 +0100
@@ -106,6 +106,7 @@
   free(status);
   displayMenu->Clear();
   cStatus::MsgOsdClear();
+  cStatus::MsgOsdMenuDestroy();
   if (!--displayMenuCount)
      DELETENULL(displayMenu);
 }
@@ -234,6 +235,7 @@
   displayMenuItems = displayMenu->MaxItems();
   displayMenu->SetTabs(cols[0], cols[1], cols[2], cols[3], cols[4]);//XXX
   displayMenu->SetTitle(title);
+  cStatus::MsgOsdMenuDisplay(menuCategory);
   cStatus::MsgOsdTitle(title);
   DisplayHelp(true);
   int count = Count();
@@ -259,13 +261,16 @@
      for (cOsdItem *item = Get(first); item; item = Next(item)) {
          bool CurrentSelectable = (i == current) && item->Selectable();
          item->SetMenuItem(displayMenu, i - first, CurrentSelectable, item->Selectable());
-         if (CurrentSelectable)
+         if (CurrentSelectable) {
             cStatus::MsgOsdCurrentItem(item->Text());
+            cStatus::MsgOsdCurrentItemIndex(item->Text(), current);
+            }
          if (++n == displayMenuItems)
             break;
          i++;
          }
      }
+  cStatus::MsgOsdMenuReady(Count());
   displayMenu->SetScrollbar(count, first);
   if (!isempty(status))
      displayMenu->SetMessage(mtStatus, status);
@@ -288,8 +293,10 @@
   cOsdItem *item = Get(current);
   if (item) {
      item->SetMenuItem(displayMenu, current - first, Current && item->Selectable(), item->Selectable());
-     if (Current && item->Selectable())
+     if (Current && item->Selectable()) {
         cStatus::MsgOsdCurrentItem(item->Text());
+        cStatus::MsgOsdCurrentItemIndex(item->Text(), current);
+        }
      if (!Current)
         item->SetFresh(true); // leaving the current item resets 'fresh'
      if (cMenuEditItem *MenuEditItem = dynamic_cast<cMenuEditItem *>(item)) {
@@ -309,8 +316,10 @@
      if (Offset >= 0 && Offset < first + displayMenuItems) {
         bool Current = Index == current;
         Item->SetMenuItem(displayMenu, Offset, Current && Item->Selectable(), Item->Selectable());
-        if (Current && Item->Selectable())
+        if (Current && Item->Selectable()) {
            cStatus::MsgOsdCurrentItem(Item->Text());
+           cStatus::MsgOsdCurrentItemIndex(Item->Text(), current);
+           }
         }
      }
 }
--- ../vdr-2.2.0.plain//status.c	2014-01-25 11:47:39.000000000 +0100
+++ status.c	2017-02-20 09:09:59.762074414 +0100
@@ -113,6 +113,12 @@
       sm->OsdCurrentItem(Text);
 }
 
+void cStatus::MsgOsdCurrentItemIndex(const char *Text, int Index)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+     sm->OsdCurrentItemIndex(Text, Index);
+}
+
 void cStatus::MsgOsdTextItem(const char *Text, bool Scroll)
 {
   for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
@@ -130,3 +136,37 @@
   for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
       sm->OsdProgramme(PresentTime, PresentTitle, PresentSubtitle, FollowingTime, FollowingTitle, FollowingSubtitle);
 }
+
+void cStatus::MsgOsdSetEvent(const cEvent* event)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+      sm->OsdSetEvent(event);
+}
+
+void cStatus::MsgOsdSetRecording(const cRecording* recording)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+      sm->OsdSetRecording(recording);
+}
+
+void cStatus::MsgOsdMenuDisplay(eMenuCategory menuCategory)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+      sm->OsdMenuDisplay(menuCategory);
+}
+
+void cStatus::MsgOsdMenuDestroy()
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+      sm->OsdMenuDestroy();
+}
+void cStatus::MsgOsdEventItem(const cEvent* Event, const char *Text, int Index, int Count)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+     sm->OsdEventItem(Event, Text, Index, Count);
+}
+void cStatus::MsgOsdMenuReady(int Count)
+{
+  for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+     sm->OsdMenuReady(Count);
+}
--- ../vdr-2.2.0.plain//status.h	2014-01-25 11:47:39.000000000 +0100
+++ status.h	2017-02-20 09:10:43.885191779 +0100
@@ -74,6 +74,8 @@
                // The OSD displays the given single line Text as menu item at Index.
   virtual void OsdCurrentItem(const char *Text) {}
                // The OSD displays the given single line Text as the current menu item.
+  virtual void OsdCurrentItemIndex(const char *Text, int Index) {}
+               // The OSD displays the given single line Text as the current menu item.
   virtual void OsdTextItem(const char *Text, bool Scroll) {}
                // The OSD displays the given multi line text. If Text points to an
                // actual string, that text shall be displayed and Scroll has no
@@ -84,6 +86,19 @@
                // The OSD displays the single line Text with the current channel information.
   virtual void OsdProgramme(time_t PresentTime, const char *PresentTitle, const char *PresentSubtitle, time_t FollowingTime, const char *FollowingTitle, const char *FollowingSubtitle) {}
                // The OSD displays the given programme information.
+  virtual void OsdSetRecording(const cRecording* recording) {}
+               // The OSD displays the recording information.
+  virtual void OsdSetEvent(const cEvent* event) {}
+               // The OSD displays the event information.
+  virtual void OsdMenuDisplay(eMenuCategory menuCategory) {}
+               // report menu creation
+  virtual void OsdMenuDestroy() {}
+               // report menu destruvtion
+  virtual void OsdEventItem(const cEvent* Event, const char *Text, int Index, int Count) {}
+               // The OSD displays the given single line Event as menu item at Index.
+  virtual void OsdMenuReady(int Count) {}
+               // The OSD menu is ready with Count Items (all Items added)
+
 public:
   cStatus(void);
   virtual ~cStatus();
@@ -103,9 +118,16 @@
   static void MsgOsdHelpKeys(const char *Red, const char *Green, const char *Yellow, const char *Blue);
   static void MsgOsdItem(const char *Text, int Index);
   static void MsgOsdCurrentItem(const char *Text);
+  static void MsgOsdCurrentItemIndex(const char *Text, int Index);
   static void MsgOsdTextItem(const char *Text,  bool Scroll = false);
   static void MsgOsdChannel(const char *Text);
   static void MsgOsdProgramme(time_t PresentTime, const char *PresentTitle, const char *PresentSubtitle, time_t FollowingTime, const char *FollowingTitle, const char *FollowingSubtitle);
+  static void MsgOsdSetEvent(const cEvent* event);
+  static void MsgOsdSetRecording(const cRecording* recording);
+  static void MsgOsdMenuDisplay(eMenuCategory menuCategory);
+  static void MsgOsdMenuDestroy();
+  static void MsgOsdEventItem(const cEvent* Event, const char *Text, int Index, int Count);
+  static void MsgOsdMenuReady(int Count);
   };
 
 #endif //__STATUS_H
