--- ../vdr-2.3.8.plain//./menu.c	2018-01-27 15:10:12.076866121 +0100
+++ ./menu.c	2018-01-27 11:19:11.560257101 +0100
@@ -5840,10 +5840,14 @@
      lastCurrent = -1; // triggers redisplay
      cStateKey StateKey;
      marks.Lock(StateKey);
-     if (cMark *m = marks.Get(Current))
+     if (cMark *m = marks.Get(Current)) {
         marks.Del(m);
+        cStatus::MsgMarkToggle(&marks, 0);
+        }
      else {
         marks.Add(Current);
+        if (cMark *m = marks.Get(Current))
+           cStatus::MsgMarkToggle(&marks, m);
         bool Play, Forward;
         int Speed;
         if (Setup.PauseOnMarkSet || GetReplayMode(Play, Forward, Speed) && !Play) {
--- ../vdr-2.3.8.plain//./status.c	2014-01-25 11:47:39.000000000 +0100
+++ ./status.c	2018-01-27 11:18:49.144100599 +0100
@@ -53,6 +53,12 @@
       sm->Replaying(Control, Name, FileName, On);
 }

+void cStatus::MsgMarkToggle(const cMarks* marks, const cMark* mark)
+{
+   for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
+      sm->MarkToggle(marks, mark);
+}
+
 void cStatus::MsgSetVolume(int Volume, bool Absolute)
 {
   for (cStatus *sm = statusMonitors.First(); sm; sm = statusMonitors.Next(sm))
--- ../vdr-2.3.8.plain//./status.h	2017-06-23 11:08:24.000000000 +0200
+++ ./status.h	2018-01-27 11:18:10.967834036 +0100
@@ -59,6 +59,8 @@
                // a name, Name can be a string that identifies the player type (like, e.g., "DVD").
                // The full file name of the recording is given in FileName, which may be NULL in case there is no
                // actual file involved. If On is false, Name may be NULL.
+  virtual void MarkToggle(const cMarks* marks, const cMark* mark) {}
+               //
   virtual void SetVolume(int Volume, bool Absolute) {}
                // The volume has been set to the given value, either
                // absolutely or relative to the current volume.
@@ -103,6 +105,7 @@
   static void MsgChannelSwitch(const cDevice *Device, int ChannelNumber, bool LiveView);
   static void MsgRecording(const cDevice *Device, const char *Name, const char *FileName, bool On);
   static void MsgReplaying(const cControl *Control, const char *Name, const char *FileName, bool On);
+  static void MsgMarkToggle(const cMarks* marks, const cMark* mark);
   static void MsgSetVolume(int Volume, bool Absolute);
   static void MsgSetAudioTrack(int Index, const char * const *Tracks);
   static void MsgSetAudioChannel(int AudioChannel);
